### ISSUE TITLE
Selects in master aren't flexing full width when they should be

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#11789]
@paulrosenzweig / @tlrobinson, I suspect that the recent change to unify our select component has introduced a problem where selects that are meant to take up all the available horizontal width aren't doing so. This is leading to alignment problems and funny gaps in layouts.

Here's the Data tab of a chart in `master`:

![image](https://user-images.githubusercontent.com/2223916/73026761-e3181480-3de6-11ea-847b-35269a1e5a18.png)

And here's EE v1.33.4.1, where you can see the width of the selects makes the icon alignment look right:

![image](https://user-images.githubusercontent.com/2223916/73026852-16f33a00-3de7-11ea-91b6-add95cd3810f.png)

Here's the Data Model in `master`:

![image](https://user-images.githubusercontent.com/2223916/73027083-9aad2680-3de7-11ea-9732-cef739e00118.png)

And here it is again in EE v1.33.4.1:

![image](https://user-images.githubusercontent.com/2223916/73027116-a7ca1580-3de7-11ea-92fb-54d59a4ace6b.png)

**Comment by tlrobinson** [comment]
Yeah, the new `Select` is not a `block` element by default, which I think is more correct (@kdoh?), but I didn't go back and find all the ones where the old behavior was correct. I think we should fix these on a case-by-case basis rather than reverting that.

Are there any other instances you've noticed it?

**Comment by mazameli** [comment]
I'll do a pass through the app and report back here.

**Comment by mazameli** [comment]
New dashboard/question:

![image](https://user-images.githubusercontent.com/2223916/73028295-15774100-3dea-11ea-96b8-e841a8a5a94b.png)

New pulse:

![image](https://user-images.githubusercontent.com/2223916/73028341-2de75b80-3dea-11ea-9cbc-fadc8d8304f7.png)

SQL variables panel:

![image](https://user-images.githubusercontent.com/2223916/73028494-80287c80-3dea-11ea-945e-30e10439643b.png)

**Comment by mazameli** [comment]
I'll take a stab at fixing these cases (provided @kdoh hasn't already).

**Comment by mazameli** [comment]
I poked at this for a little, and what I'm unclear on is how to pass down to `SelectButton` a class name or prop that will get it to fill the available space, short of adding `flex-full` to `SelectButton` itself (which fixes all the screenshots I showed, but then makes the default select really wide in places like the add-DB form).

I say this because It looks like the `Select` parent component is already full width:

![image](https://user-images.githubusercontent.com/2223916/73031846-1318e500-3df2-11ea-8e05-d2e66beef112.png)

… but `SelectButton` isn't:

![image](https://user-images.githubusercontent.com/2223916/73031871-2f1c8680-3df2-11ea-8339-f6a4c06a00fa.png)

I tried a few things in `ChartSettingSelect` like mucking around with the classNames in the Select there, but to no avail. The Select in `ChartSettingSelect` _was_ already set to `block flex-auto`, but it doesn't appear to be doing what's intended.
